pkg/registry/bundle.go

```diff
@@ -297,19 +320,39 @@ func (b *Bundle) cache() error {
 			}
 			dec := serializer.NewCodecFactory(Scheme).UniversalDeserializer()
 			switch o.GroupVersionKind().Version {
-			case v1CRDVersion:
+			case v1Version:
 				crd := &apiextensionsv1.CustomResourceDefinition{}
 				if _, _, err = dec.Decode(cb, nil, crd); err != nil {
 					return fmt.Errorf("error decoding v1 CRD: %v", err)
 				}
 				b.v1crds = append(b.v1crds, crd)
-			case v1beta1CRDVersion:
+			case v1beta1Version:
 				crd := &apiextensionsv1beta1.CustomResourceDefinition{}
 				if _, _, err = dec.Decode(cb, nil, crd); err != nil {
 					return fmt.Errorf("error decoding v1beta1 CRD: %v", err)
 				}
 				b.v1beta1crds = append(b.v1beta1crds, crd)
 			}
+		case APIServiceKind:
```

Comment:
IIRC, `APIServices` are only used to infer the `APIServiceDescriptions` field on generated CSV's. If they are pulled into the `Objects` field, we need to make sure they are excluded when the bundle is serialized for the database, otherwise we could end up with `APIServices` in the content that gets applied to the cluster (which isn't what we want).

Reply:
These are provided APIService, shouldn't they be applied to cluster? (we are not using any of the resources here to apply to the cluster. Not my intention anyway.)

Reply:
No -- OLM generates and applies APIServices based on the content of the CSV. These should be used to synthesize a CSV.

Reply:
I forgot this was the work I needed when we are merging apiSevirce and CRDs. This is used in Validation work to remove delineation between CRD and APIService. Since we are not merging that OLM PR, this should be removed then.

Reply:
removed it as a separate commit